File: cpp-examples/composition/BookShelf.cpp
#include "BookShelf.h"
#include <iostream>

BookShelf::BookShelf() {}

BookShelf::~BookShelf() {
    for (auto section : sections) {
        delete section;
    }
    sections.clear();
}

void BookShelf::addSection(Section* section) {
    sections.push_back(section);
}

void BookShelf::printDetails() const {
    std::cout << "BookShelf Details:" << std::endl;
    for (const auto& section : sections) {
        section->printDetails();
    }
}

File: python-examples/composition/BookShelf.py
from Section import Section
from Science import Science
from Fiction import Fiction

class BookShelf:
    def __init__(self):
        self.sections = []

    def add_section(self, section):
        self.sections.append(section)

    def print_details(self):
        print("BookShelf Details:")
        for section in self.sections:
            section.print_details()

